tm2py/components/network/transit/transit_assign.py

```diff
@@ -1,9 +1,632 @@
+            "boarding_time": {
+                "global": {
+                    "penalty": self._transit_config.initial_boarding_penalty,
+                    "perception_factor": 1,
+                }
+            },
+            "in_vehicle_cost": None,
+            "in_vehicle_time": {"perception_factor": "@invehicle_factor"},
+            "aux_transit_time": {
+                "perception_factor": self._transit_config.walk_perception_factor
+            },
+            "aux_transit_cost": None,
+            "journey_levels": self._journey_levels,
+            "flow_distribution_between_lines": {"consider_total_impedance": False},
+            "flow_distribution_at_origins": {
+                "fixed_proportions_on_connectors": None,
+                "choices_at_origins": "OPTIMAL_STRATEGY",
+            },
+            "flow_distribution_at_regular_nodes_with_aux_transit_choices": {
+                "choices_at_regular_nodes": "OPTIMAL_STRATEGY"
+            },
+            "circular_lines": {"stay": False},
+            "connector_to_connector_path_prohibition": None,
+            "od_results": {"total_impedance": None},
+            "performance_settings": {"number_of_processors": self._num_processors},
+        }
+        if self._transit_config.use_fares:
+            fare_perception = 60 / self._transit_config.value_of_time
+            spec["boarding_cost"] = {
+                "on_segments": {
+                    "penalty": "@board_cost",
+                    "perception_factor": fare_perception,
+                }
+            }
+            spec["in_vehicle_cost"] = {
+                "penalty": "@invehicle_cost",
+                "perception_factor": fare_perception,
+            }
+        # Optional aux_transit_cost, used for walk time on connectors,
+        #          set if override_connector_times is on
+        if self._transit_config.get("override_connector_times", False):
+            spec["aux_transit_cost"] = {
+                "penalty": f"@walk_time_{self.name.lower()}",
+                "perception_factor": self._transit_config.walk_perception_factor,
+            }
+        return spec
+
+    @property
+    def _demand_matrix(self) -> str:
+        if self._iteration < 1:
+            return 'ms"zero"'  # zero demand matrix
+        return f'mf"TRN_{self._class_config.skim_set_id}_{self._time_period}"'
+
+    def _get_used_mode_ids(self, modes: List[TransitModeConfig]) -> List[str]:
+        """Get list of assignment Mode IDs from input list of Emme mode objects.
+
+        Accounts for fare table (mapping from input mode ID to auto-generated
+        set of mode IDs for fare transition table (fares.far input) by applyfares
+        component.
+        """
+        if self._transit_config.use_fares:
+            out_modes = set([])
+            for mode in modes:
+                if mode.assign_type == "TRANSIT":
+                    out_modes.update(self._fare_modes[mode.mode_id])
```

Comment:
the assignment of something to be used in fare calculation shiould be done separately - not hidden in this method